data/data-pipeline/data_pipeline/etl/score/etl_utils.py

```diff
@@ -305,3 +312,89 @@ def create_codebook(
     return merged_codebook_df[constants.CODEBOOK_COLUMNS].rename(
         columns={constants.CEJST_SCORE_COLUMN_NAME: "Description"}
     )
+
+
+# pylint: disable=too-many-arguments
+def compare_to_list_of_expected_state_fips_codes(
+    actual_state_fips_codes: typing.List[str],
+    nation_expected: bool = True,
+    puerto_rico_expected: bool = True,
+    island_areas_expected: bool = True,
+    additional_fips_codes_not_expected: typing.List[str] = None,
+    dataset_name: str = None,
+) -> None:
+    """Check whether a list of state/territory FIPS codes match expectations.
+
+    Args:
+        actual_state_fips_codes (List of str): Actual state codes observed in data
+        nation_expected (bool, optional): Do you expect the nation (DC & states) to be
+            represented in data?
+        puerto_rico_expected (bool, optional): Do you expect PR to be represented in data?
+        island_areas_expected (bool, optional): Do you expect Island Areas to be represented in
+            data?
+        additional_fips_codes_not_expected (List of str, optional): Additional state codes
+            not expected in the data. For example, the data may be known to be missing
+            data from Maine and Wisconsin.
+        dataset_name (str, optional): The name of the data set, used only in printing an
+            error message. (This is helpful for debugging during parallel etl runs.)
+
+    Returns:
+        None: Does not return any values.
+
+    Raises:
+        ValueError: if lists do not match expectations.
+    """
+    # Setting default argument of [] here to avoid mutability problems.
+    if additional_fips_codes_not_expected is None:
+        additional_fips_codes_not_expected = []
+
+    # Cast input to a set.
+    actual_state_fips_codes_set = set(actual_state_fips_codes)
+
+    # Start with the list of all FIPS codes for all states and territories.
+    expected_states_set = set(get_state_fips_codes(settings.DATA_PATH))
+
+    # If nation (states and DC) are not expected to be included, remove it from the
+    # expected
+    # states set.
+    if not nation_expected:
+        expected_states_set = expected_states_set - set(TILES_NATION_FIPS_CODE)
+
+    # If Puerto Rico is not expected to be included, remove it from the expected
+    # states set.
+    if not puerto_rico_expected:
+        expected_states_set = expected_states_set - set(
+            TILES_PUERTO_RICO_FIPS_CODE
+        )
+
+    # If island areas are not expected to be included, remove them from the expected
+    # states set.
+    if not island_areas_expected:
+        expected_states_set = expected_states_set - set(
+            TILES_ISLAND_AREA_FIPS_CODES
+        )
+
+    # If additional FIPS codes are not expected to be included, remove them from the
+    # expected states set.
+    expected_states_set = expected_states_set - set(
+        additional_fips_codes_not_expected
+    )
+
+    dataset_name_phrase = (
```

Comment:
I low key don't think this is necessary and it just adds a lot of text -- most of the classes don't have a dataset name right now!

Reply:
and, if you want to keep it, it should have a space so that my comment about spaces becomes irrelevant

Reply:
The name logging actually was very helpful debugging *which* ETL classes were failing when they were running in parallel. I was just seeing a bunch of "these states are not as expected!" but because the logs were jumbled from parallel runs, I had no idea which ETL was failing.

Reply:
i see i see, i am sold